pkg/kubelet/dockershim/network/cni/cni.go

```diff
@@ -192,14 +195,22 @@ func getDefaultCNINetwork(confDir string, binDirs []string) (*cniNetwork, error)
 			continue
 		}
 
+		// Before using this CNI config, we have to validate it to make sure that
+		// all plugins of this config exist on disk
+		caps, err := cniConfig.ValidateNetworkList(context.TODO(), confList)
+		if err != nil {
+			klog.Warningf("Error validating CNI config %v: %v", confList, err)
+			continue
```

Comment:
I was chatting with @dcbw and we think that, if the lowest-named configuration is not valid, we shouldn't continue on to the next file. Instead, we should just abort and wait for the situation to be fixed.

I realize that every other check just does a `continue`, but this is the only failure that can happen because of outside factors. For example ValidateNetworkList makes sure that every plugin binary is installed on disk.

Maybe someday we should remove most of the other `continue`s as well - if someone writes 000-foo.conflist, and they have a typo, we should probably stop there, as their intent is pretty clear.

Reply:
I agree with this, @squeed , this will make things clear when we have several CNI config files to choose. Should we have an update here, or pull another PR after some detailed consideration?

Reply:
Hmm, probably just this PR is enough for now :-)